rust/protocol/src/group_cipher.rs

```diff
@@ -40,15 +40,22 @@ pub async fn group_encrypt<R: Rng + CryptoRng>(
 
     let message_keys = sender_chain_key.sender_message_key();
 
-    let ciphertext =
-        crypto::aes_256_cbc_encrypt(plaintext, message_keys.cipher_key(), message_keys.iv())
-            .map_err(|_| {
-                log::error!(
-                    "outgoing sender key state corrupt for distribution ID {}",
-                    distribution_id,
-                );
-                SignalProtocolError::InvalidSenderKeySession { distribution_id }
-            })?;
+    let cipher_key: [u8; crypto::AES_256_KEY_SIZE] =
+        message_keys.cipher_key().try_into().map_err(|_| {
+            log::error!(
+                "outgoing sender key state corrupt for distribution ID {}",
+                distribution_id,
+            );
+            SignalProtocolError::InvalidSenderKeySession { distribution_id }
+        })?;
+    let iv: [u8; crypto::AES_NONCE_SIZE] = message_keys.iv().try_into().map_err(|_| {
+        log::error!(
+            "outgoing sender key state corrupt for distribution ID {}",
+            distribution_id,
+        );
+        SignalProtocolError::InvalidSenderKeySession { distribution_id }
+    })?;
+    let ciphertext = crypto::cbc::aes_256_cbc_encrypt(plaintext, &cipher_key, &iv);
```

Comment:
This duplication is a little annoying, but I can't see a great way to clean it up. Do you think it's worth putting that error generation into a local helper?